### ISSUE TITLE
SQLFluff fails to properly parse UNNEST which has "WITH OFFSET" clause.

### ISSUE THREAD

**Issue by stas-snow** [sqlfluff/sqlfluff#2787]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff fails to properly parse UNNEST which has "WITH OFFSET" clause which is part of BigQuery sql language.

### Expected Behaviour

I would expect that sqlfluff would properly parse BigQuery dialect and properly handle WITH OFFSET inside of UNNEST statement.

### Observed Behaviour

Found unparsable section:

### How to reproduce

Run command:
```
sqlfluff parse --dialect bigquery test.sql
```

test.sql:
```
	SELECT id, name
	FROM
		UNNEST([1, 2, 3]) id WITH OFFSET pos1,
		UNNEST(['a', 'b', 'c']) name WITH OFFSET pos2
	WHERE pos1 = pos2
```

Output:
```
L:   3 | P:  38 |  PRS | Line 3, Position 38: Found unparsable section: "pos1,\n
                       | UNNEST(['a', 'b', 'c']) name W..."
```

### Dialect

bigquery

### Version

sqlfluff, version 0.10.1

### Configuration

.sqlfluff

```
[sqlfluff]
# Exclude the following rules:
# L034: Select wildcards then simple targets before calculations and aggregates.
exclude_rules = L034

# Some rules can be configured directly from the config common to other rules
[sqlfluff:rules]
# Default max_line_length = 80
max_line_length = 120
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
We currently support this:

```sql
UNNEST(['a', 'b', 'c']) WITH OFFSET pos2
```

But not this:

```sql
UNNEST(['a', 'b', 'c']) name WITH OFFSET pos2
```

#2793 adds this support and will be in 0.11.0 which we should release today.

**Comment by stas-snow** [comment]
Thank you for fixing this!
